flink-runtime/src/main/java/org/apache/flink/runtime/asyncprocessing/AsyncExecutionController.java

```diff
@@ -189,6 +194,9 @@ public RecordContext<K> buildContext(Object record, K key) {
      */
     public void setCurrentContext(RecordContext<K> switchingContext) {
         currentContext = switchingContext;
+        if (switchContextListener != null) {
```

Comment:
When will be `switchContextListener` be null?
This is on the critical path, is it okay not to check for null?

Reply:
for normal async statebackend  (like forst), this could be null since there is nothing to do with statebackend.

Reply:
If we keep a empty listener here instead of a `null`, the influence on the hot path for `async` is similar (`if` vs. a method call). For sync path (a rocksdb wrapped with adaptor), the `setCurrentContext` will be only invoked once, since the context switch is skipped for any callbacks of `CompletedStateFuture`. Thus the influence is minor, I'd prefer keeping it.